api-admin/src/main/java/bio/terra/pearl/api/admin/service/siteContent/SiteImageExtService.java

```diff
@@ -34,4 +34,20 @@ public List<SiteImageMetadata> list(String portalShortcode, AdminUser operator)
     authUtilService.authUserToPortal(operator, portalShortcode);
     return siteImageService.findMetadataByPortal(portalShortcode);
   }
+
+  public SiteImage upload(
+      String portalShortcode, String uploadFileName, byte[] imageData, AdminUser operator) {
+    authUtilService.authUserToPortal(operator, portalShortcode);
+    String cleanFileName = SiteImageService.cleanFileName(uploadFileName);
+    int version = siteImageService.getNextVersion(cleanFileName, portalShortcode);
+    SiteImage image =
+        SiteImage.builder()
+            .portalShortcode(portalShortcode)
+            .version(version)
+            .data(imageData)
+            .uploadFileName(uploadFileName)
```

Comment:
Feels a bit cleaner if we populate the `cleanFileName` here?
Also, why do we store the `uploadFileName"?

Reply:
ah, yes, you asked about that in the last PR.  Because I'm a data hoarder?  :).  I'll make a ticket to remove it